Tasks/DownloadPackage/download.ts

```diff
@@ -50,8 +50,9 @@ export async function downloadPackage(collectionUrl: string, credentialHandler:
 	
 	await new Promise((resolve, reject) => {
 		feedConnection.getCoreApi().restClient.get(packageUrl, ApiVersion, null, { responseIsCollection: false }, async function (error, status, result) {
-			if (!!error && status != 200) {
-				reject(tl.loc("FailedToGetPackageMetadata", error));
+			if (!!error || status != 200) {
+				tl.setResult(tl.TaskResult.Failed, tl.loc("FailedToGetPackageMetadata", error));
```

Comment:
do we need to set tl.setResult before every return? wouldn't the tl.setResult in main handle setting the result?